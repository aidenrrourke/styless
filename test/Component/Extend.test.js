import React from 'react';
import renderer from 'react-test-renderer';
import styled from 'styled-components';
import 'jest-styled-components';

test('Extend a styled div', () => {
    const Div = styled.div`
		color: @primary;
	`;
    const Div2 = Div.extend`
		color: @primary;
	`;
    expect(renderer.create(<Div primary="red"/>).toJSON()).toMatchSnapshot();
    expect(renderer.create(<Div2 primary="red"/>).toJSON()).toMatchSnapshot();
});