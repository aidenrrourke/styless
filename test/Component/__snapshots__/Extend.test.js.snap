// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Extend a styled div 1`] = `
.c0 {
  color: red;
}

<div
  className="c0"
/>
`;

exports[`Extend a styled div 2`] = `
.c0 {
  color: red;
  color: red;
}

<div
  className="c0"
/>
`;
